flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/runtime/operators/over/NonTimeUnboundedPrecedingFunctionTest.java

```diff
@@ -0,0 +1,467 @@
+
+    private static final GeneratedRecordComparator GENERATED_SORT_KEY_COMPARATOR =
+            new GeneratedRecordComparator("", "", new Object[0]) {
+
+                private static final long serialVersionUID = 1L;
+
+                @Override
+                public RecordComparator newInstance(ClassLoader classLoader) {
+                    return new LongRecordComparator(SORT_KEY_IDX);
+                }
+            };
+
+    private static final GeneratedRecordEqualiser GENERATED_RECORD_EQUALISER =
+            new GeneratedRecordEqualiser("", "", new Object[0]) {
+
+                private static final long serialVersionUID = 1L;
+
+                @Override
+                public RecordEqualiser newInstance(ClassLoader classLoader) {
+                    // return new RowDataRecordEqualiser();
+                    return new TestRecordEqualiser();
+                }
+            };
+
+    /** Custom test comparator for comparing numbers. */
+    public static class LongRecordComparator implements RecordComparator {
+
+        private int pos;
+
+        public LongRecordComparator(int pos) {
+            this.pos = pos;
+        }
+
+        @Override
+        public int compare(RowData o1, RowData o2) {
+            boolean null0At1 = o1.isNullAt(pos);
+            boolean null0At2 = o2.isNullAt(pos);
+            int cmp0 =
+                    null0At1 && null0At2
+                            ? 0
+                            : (null0At1
+                                    ? -1
+                                    : (null0At2
+                                            ? 1
+                                            : Long.compare(o1.getLong(pos), o2.getLong(pos))));
+            if (cmp0 != 0) {
+                return cmp0;
+            }
+            return 0;
+        }
+    }
+
+    /** Custom test row equaliser for comparing rows. */
+    public static class TestRecordEqualiser implements RecordEqualiser {
+
+        private static final long serialVersionUID = -6706336100425614942L;
+
+        @Override
+        public boolean equals(RowData row1, RowData row2) {
+            if (row1 instanceof BinaryRowData && row2 instanceof BinaryRowData) {
+                return row1.equals(row2);
+            } else if (row1 instanceof GenericRowData && row2 instanceof GenericRowData) {
+                return row1.getString(0).equals(row2.getString(0))
+                        && row1.getLong(1) == row2.getLong(1)
+                        && row1.getLong(2) == row2.getLong(2);
+            } else {
+                throw new UnsupportedOperationException();
+            }
+        }
+    }
+
+    @Test
+    public void testInsertOnlyRecordsWithCustomSortKey() throws Exception {
+        NonTimeUnboundedPrecedingFunction<RowData> function =
+                new NonTimeUnboundedPrecedingFunction<RowData>(
+                        0,
+                        2000,
+                        aggsHandleFunction,
```

Comment:
`aggsHandleFunction` uses a `SUM` function. We should either change this to `AVG` or add a new `avgAggsHandleFunction` for our tests.
In contrast to the `SUM` function, the `AVG` function uses different types for the accumulator and result value.
Using `AVG` should have caught all issues in which accumulators and values were used interchangeably.

Reply:
Added a test for AVG. However, using accumulators and values interchangeably worked. For consistency, I have made the change to use accumulators while comparing and storing.

Reply:
Added a test for AVG as a restore test